test/systemtests/k8setup_test.go

```diff
@@ -439,21 +439,26 @@ func (k *kubernetes) cleanupContainers() error {
 	return nil
 }
 
-func (k *kubernetes) startNetplugin(args string) error {
-	if k.node.Name() == "k8master" {
-		return nil
-	}
+func (k *kubernetes) commonArgs() string {
 	netMode := k.node.suite.globInfo.Encap
 	fwdMode := k.node.suite.fwdMode
+	mode := "kubernetes"
 	var storeArgs string
-	if cStore := k.node.suite.basicInfo.ClusterStore; strings.HasPrefix(cStore, "etcd") {
-		storeArgs = " --etcd-endpoints " + strings.Replace(cStore, "etcd", "http", 1) + " "
+	if k.node.suite.basicInfo.ClusterStoreDriver == "etcd" {
+		storeArgs = " --etcd-endpoints " + k.node.suite.basicInfo.ClusterStoreURLs + " "
 	} else {
-		storeArgs = " --consul-endpoints " + strings.Replace(cStore, "consul", "http", 1) + " "
+		storeArgs = " --consul-endpoints " + k.node.suite.basicInfo.ClusterStoreURLs + " "
+	}
+	return " --netmode " + netMode + " --fwdmode " + fwdMode + " --mode " + mode + storeArgs
+}
```

Comment:
is the only diff with docker setup tset is the mode? reuse?

Reply:
it's due to the way how our sys testing was done, `k8setpup_test.go` and `docker_test.go` are separated scheduler and codes between them are not able to share, to fix this we need to introduce a new util and generates shared codes ... i'd leave it in different PR